beesdoo_easy_my_coop/models/res_partner.py

```diff
@@ -0,0 +1,44 @@
+from odoo import models, fields, api, _
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    cooperator_type = fields.Selection(selection='_get_share_type', compute='_compute_share_type', string='Cooperator Type', store=True)
+    can_shop = fields.Boolean(compute='_can_shop', store=True)
+    info_session_confirmed = fields.Boolean(
+        string="Confirmed presence to info session",
+        default=False,
+    )
+
+    @api.depends('cooperator_type', 'cooperative_status_ids', 'cooperative_status_ids.can_shop')
+    def _can_shop(self):
+        for rec in self:
+            if rec.cooperator_type == 'share_b':
```

Comment:
this should not be hardcoded anymore. a boolean can_shop on the share(product.template) should do the work

Reply:
There is already a flag in easy_my_coop share product named `customer` that do the same thing then `can_shop`. So I've removed `can_shop` on the share product.

See : https://github.com/coopiteasy/vertical-cooperative/blob/12.0/easy_my_coop/models/product.py#L21

Reply:
as far that the can_shop on res_partner is handled and correctly mapped to customer field from the product, it is fine for me.